src/backend/app.js

```diff
@@ -18,8 +19,85 @@ app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
 
 app.use(cors());
+const knex = require('knex')({
+  client :'mysql2',
+  connection: {
+      host : process.env.DB_HOST,
+      //@ts-ignorets-
+      port : process.env.DB_PORT,
+      user : process.env.DB_USER,
+      password: process.env.DB_PASSWORD,
+      database:process.env.DB_NAME,
+  }
+   })
 
 router.use("/meals", mealsRouter);
+// 1.Respond with all meals in the future (relative to the when datetime)
+router.use("/future-meals", async (request, response) => {
+ 
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const rows= await knex.raw("select * from Meal WHERE Meal.when > now()");
+    response.json(rows[0]);
+   } catch (error) {
+    throw error;
+  }
+}),
+
+// 2.Respond with all meals in the past (relative to the when datetime)
+router.use("/past-meals", async (request, response) => {
+ 
+  try {
+    const rows= await knex.raw("select * from Meal WHERE Meal.when < now()");
+    response.json(rows[0]);
+   } catch (error) {
+    throw error;
+  }
+}),
+
+// 3.Respond with all meals sorted by ID
+router.use("/all-meals", async (request, response) => {
+ 
+  try {
+    //const rows = await knex.raw("select MIN(id) from Meal")
+    const rows= await knex.raw("select * from Meal ORDER BY id");
+    response.json(rows[0]);
+   } catch (error) {
+    throw error;
+  }
+}),
+
+// 4.Respond with the first meal (meaning with the minimum id)
+router.use("/first-meal", async (request, response) => {
+ 
+  try {
+    //const rows = await knex.raw("select MIN(id) from Meal")
+    const rows= await knex.raw("select * from Meal WHERE id  LIMIT 1");
+    response.json(rows[0]);
+   } catch (error) {
+    throw error;
+  }
+}),
+
+// 5.Respond with the last meal (meaning with the maximum id)
+router.use("/last-meal", async (request, response) => {
+ 
+  try{
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const rows= await knex.raw(" select * from Meal ORDER BY id DESC LIMIT 1");
```

Comment:
May be a good idea to name vairable as row and not rows when you know that there will be only one record returned by query

Reply:
@itsgauravsaxena  Thank you, Sure it makes sense :)